### PR TITLE
Add template for 404 page

### LINKED ISSUE #11: Update 404 template
Update the 404 template so that it matches the [desktop](https://www.figma.com/file/nf4oV51c1JBxJKXshMe7cd/New-Sensei-Theme?node-id=20020078%3A4324) and [mobile](https://www.figma.com/file/nf4oV51c1JBxJKXshMe7cd/New-Sensei-Theme?node-id=20020126%3A4533) designs. Example PR [here](https://github.com/Automattic/themes/pull/6505).

## Desktop
Screen Shot 2022-09-30 at 8 30 22 AM

## Mobile
Screen Shot 2022-09-30 at 8 31 01 AM

#### Scenario: 404 template
**Given** The Sensei theme is activated
**When** I view a page that does not exist
**Then** the page looks similar to the design on both desktop and mobile.

### PR DESCRIPTION
**Pull request by Imran92**
Implements https://github.com/Automattic/sensei-theme/issues/11

## Changes proposed
- Updated the template and the pattern for the 404 page to match the design

## Testing instructions

- In your browser, hit your sensei site with a URL that does not exist ( like http://localsensei.test/random-url-that-does-not-exist)
- Make sure the page matches the design

### Desktop

image

### Mobile

image

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -365,4 +366,21 @@ body .is-layout-flow > * {
 
 body .is-layout-constrained > * + * {
 	margin-block-start: 0px;
+}
+
+/* Design for search box in 404 page template sidebar */
+.course-search-sidebar button {
```
Comment (gabrielcaires) [inline style.css:372]: We already have a search bar done on the search template. Can we reuse it?

**Review by gabrielcaires — CHANGES_REQUESTED** [review]
The search button is not working
There is an implementation on the search template.

File: style.css (hunk 1 above)
Comment (Imran92) [reply style.css:372]: Thanks for the suggestion @gabrielcaires <3 I couldn't reuse the style in the search template the search button there is added using CSS and SVG file, the style added for the search button on search template depends on the search button being empty, but I used the built in block settings for the icon. So I added the style. Is there any special reason why we didn't use the icon setting in the search template? I used it this way -

https://user-images.githubusercontent.com/6820724/199707736-d060a23c-f2d4-4134-b1b9-2987a5958b75.mov

**Comment by Imran92** [comment]
> The search button is not working There is an implementation on the search template.

Thanks for checking @gabrielcaires , I just tried and it's working for me locally. Can you kindly refresh the cache and try again? Or is there any specific condition that I might be missing?

https://user-images.githubusercontent.com/6820724/199708241-c43b1274-b312-48a0-80f3-479f366e8ad0.mov

**Review by gabrielcaires — APPROVED** [review]
@Imran92 Yes, I don't why but now it is working :( 
I just found a small adjustment, the search bar + button should have 56px instead of 58px.
Screenshot 2022-11-04 at 14 00 02

**Pull request merged**